### PR TITLE
Bug fix/ask lisa related fixes

### PR DESCRIPTION
**Pull request by AmanPrasad43**
Thank you for your contribution.
Before submitting this PR, please make sure:

- [ ] PR description and commit message should describe the changes done in this PR
- [ ] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [ ] Latest Code from master or specific release branch is merged to your branch
- [ ] No unwanted\commented\junk code is included
- [ ] No new warning upon build solution
- [ ] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [ ] After creating pull request there should not be any conflicts
- [ ] Resolve all Codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Update the Help Library document to match any feature changes

## Summary by CodeRabbit

- **New Features**
  - Enhanced encryption handling for sensitive configuration data.
  - Improved chatbot initialization to optimize resource management and token acquisition.

- **Improvements**
  - Refined the initialization process and conditional checks for enabling the chatbot feature.

- **Bug Fixes**
  - Fixed issues with configuration management ensuring correct values are used from the solution profile.

### CODE REVIEW

File: Ginger/GingerCoreNET/GenAIServices/GenAIServiceHelper.cs [hunk 1]
```diff
@@ -162,7 +171,23 @@ public async Task<string> StartNewChat(string chatBotRequest)
         }
 
 
+        private static string CredentialsCalculation(string value)
+        {
+            ValueExpression valueExpression = new();
```
Comment (Maheshkale447) [inline Ginger/GingerCoreNET/GenAIServices/GenAIServiceHelper.cs:176]: Consider using singleton pattern here, so vale expression object will be initialized once and reused afterwards

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

File: Ginger/GingerCoreNET/GenAIServices/GenAIServiceHelper.cs [hunk 2]
```diff
@@ -162,7 +171,23 @@ public async Task<string> StartNewChat(string chatBotRequest)
...
+                return value;
+            }
+            else if (EncryptionHandler.IsStringEncrypted(value))
+            {
+                value = EncryptionHandler.DecryptwithKey(value);
+                return value;
+            }
+
+            return value;
+        }
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/GingerCoreNET/GenAIServices/GenAIServiceHelper.cs]: > The `CredentialsCalculation` method is well-implemented, handling both value expressions and encrypted strings. Consider adding inline comments to explain the logic for future maintainability.

suggest comment

File: Ginger/Ginger/ExternalConfigurations/AskLisaConfigurationsPage.xaml.cs [hunk 3]
```diff
@@ -88,5 +89,23 @@
...
+            }
+        }
+
+        private void xClientSecretLabelTextBox_LostKeyboardFocus(object sender, System.Windows.Input.KeyboardFocusChangedEventArgs e)
+        {
+            if (!EncryptionHandler.IsStringEncrypted(xClientSecretLabelTextBox.ValueTextBox.Text))
+            {
+                xClientSecretLabelTextBox.ValueTextBox.Text = ValueExpression.IsThisAValueExpression(xClientSecretLabelTextBox.ValueTextBox.Text) ? xClientSecretLabelTextBox.ValueTextBox.Text : EncryptionHandler.EncryptwithKey(xClientSecretLabelTextBox.ValueTextBox.Text);
+
+            }
```
[the remark this answers is not in the document]
Comment (AmanPrasad43) [reply Ginger/Ginger/ExternalConfigurations/AskLisaConfigurationsPage.xaml.cs:108]: > Similar to the previous comment, the encryption handling in `xClientSecretLabelTextBox_LostKeyboardFocus` ensures sensitive data is stored securely. Again, a comment explaining the conditional logic would enhance maintainability.

suggest comment

File: Ginger/Ginger/UserControlsLib/ChatBot/ChatbotWindow.xaml.cs [hunk 4]
```diff
@@ -70,9 +69,9 @@ private async Task SendMessageToAPI()
             ShowLoader();
             try
             {
-                //need to handle as we are getting token everytime
 
-                
+                genAIServiceHelper = new GenAIServiceHelper();
```
Comment (Maheshkale447) [inline Ginger/Ginger/UserControlsLib/ChatBot/ChatbotWindow.xaml.cs:73]: its not recommended to create this each time user sends a message. please initiate it once, and re-initialize only when its null

**Review by Maheshkale447 — CHANGES_REQUESTED** [review]

**Pull request merged**